budgetportal/sitemaps.py

```diff
@@ -0,0 +1,184 @@
+from django.contrib import sitemaps
+from django.urls import reverse
+from budgetportal.summaries import get_consolidated_expenditure_treemap
+from guide_data import category_guides
+from .models import (
+    InfrastructureProjectPart,
+    FinancialYear,
+    Department,
+    EXPENDITURE_TIME_SERIES_PHASE_MAPPING,
+    Government,
+)
+
+
+class ConsolidatedJsonViewSitemap(sitemaps.Sitemap):
+    def items(self):
+        years = []
+        for year in FinancialYear.get_available_years():
+            years.append(str(year.slug))
+
+        return years
+
+    def location(self, item):
+        return reverse("consolidated", args=[item])
+
+
+class DepartmentListViewSitemap(sitemaps.Sitemap):
+    def items(self):
+        years = []
+        for year in FinancialYear.get_available_years():
+            years.append(str(year.slug))
+
+        return years
+
+    def location(self, item):
+        return reverse("department-list", args=[item])
+
+
+class ProvincialDepartmentsSitemap(sitemaps.Sitemap):
+    def items(self):
+        return Department.objects.filter(government__sphere__slug="provincial")
+
+    def location(self, item):
+        return reverse(
+            "provincial-department",
+            args=[
+                item.government.sphere.financial_year.slug,
+                item.government.sphere.slug,
+                item.government.slug,
+                item.slug,
+            ],
+        )
+
+
+class NationalDepartmentsSitemap(sitemaps.Sitemap):
+    def items(self):
+        return Department.objects.filter(government__sphere__slug="national")
+
+    def location(self, item):
+        return reverse(
+            "national-department",
+            args=[item.government.sphere.financial_year.slug, item.slug],
+        )
+
+
+class DepartmentPreviewSitemap(sitemaps.Sitemap):
+    def items(self):
+        return Department.objects.all()
+
+    def location(self, item):
+        return reverse(
+            "department-preview",
+            args=[
+                item.government.sphere.financial_year.slug,
+                item.government.sphere.slug,
+                item.government.slug,
+                item.slug,
+            ],
+        )
+
+
+class DepartmentPreviewJsonSitemap(sitemaps.Sitemap):
```

Comment:
Please remove this class.
Only want sitemap for HTML endpoints, not .json.

Reply:
removed

https://github.com/vulekamali/datamanager/pull/185/commits/a09998b7e7d312745667d4696eeaa0878a379133